### PR TITLE
Feat/bending requirements

### LINKED ISSUE #23: Build code requirement for bending capacity

### PR DESCRIPTION
**Pull request by ImMPrada**
Closes #23 

This PR solves bending reinforcement ratio solicitation, solving the equation:

image

And attempting the constraint of a minimum ratio of 0.0025 for footings

### CODE REVIEW

File: lib/structuraid_core/design_codes/aci_318_19/rc/footings/bending_reinforcement_ratio.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            end
+
+            def equation_component_a
+              0.59 * design_steel_yield_strength / design_compression_strength
+            end
+
+            def equation_component_c
+              section_area = width * effective_height # m**2
+              reduced_steel_strength = capacity_reduction_factor * design_steel_yield_strength # N/(m**2)
+              flexural_moment / (reduced_steel_strength * section_area * effective_height) # N/(m**2)
```
Comment (mapra99) [inline lib/structuraid_core/design_codes/aci_318_19/rc/footings/bending_reinforcement_ratio.rb:46]: I think the unit comments are wrong
```suggestion
              section_area = width * effective_height # mm**2
              reduced_steel_strength = capacity_reduction_factor * design_steel_yield_strength # MPa
              flexural_moment / (reduced_steel_strength * section_area * effective_height) # MPa
```

File: lib/structuraid_core/design_codes/aci_318_19/rc/footings/bending_reinforcement_ratio.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            end
+
+            def equation_component_root
+              unless 4 * equation_component_a * equation_component_c < 1
+                raise RequirementNotFulfilledError.new(
+                  :flexural_moment,
+                  "Moment #{flexural_moment} is to hight for this element",
```
Comment (mapra99) [inline lib/structuraid_core/design_codes/aci_318_19/rc/footings/bending_reinforcement_ratio.rb:53]:
```suggestion
                  "Moment #{flexural_moment} is too high for this element",
```

File: lib/structuraid_core/design_codes/nsr_10/rc/footings/bending_reinforcement_ratio.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+            def equation_component_a
+              0.59 * design_steel_yield_strength / design_compression_strength
+            end
+
+            def equation_component_c
+              section_area = width * effective_height # m**2
+              reduced_steel_strength = capacity_reduction_factor * design_steel_yield_strength # N/(m**2)
+              flexural_moment / (reduced_steel_strength * section_area * effective_height) # N/(m**2)
+            end
```
Comment (mapra99) [inline lib/structuraid_core/design_codes/nsr_10/rc/footings/bending_reinforcement_ratio.rb:48]:
```suggestion
            def equation_component_c
              section_area = width * effective_height # mm**2
              reduced_steel_strength = capacity_reduction_factor * design_steel_yield_strength # MPa
              flexural_moment / (reduced_steel_strength * section_area * effective_height) # MPa
            end
```

File: lib/structuraid_core/design_codes/nsr_10/rc/footings/bending_reinforcement_ratio.rb [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+            def equation_component_root
+              unless 4 * equation_component_a * equation_component_c < 1
+                raise RequirementNotFulfilledError.new(
+                  :flexural_moment,
+                  "Moment #{flexural_moment} is to hight for this element",
```
Comment (mapra99) [inline lib/structuraid_core/design_codes/nsr_10/rc/footings/bending_reinforcement_ratio.rb:54]:
```suggestion
                  "Moment #{flexural_moment} is too high for this element",
```

File: lib/structuraid_core/design_codes/schemas/rc/footings/bending_reinforcement_ratio_schema.rb
Comment (mapra99) [inline lib/structuraid_core/design_codes/schemas/rc/footings/bending_reinforcement_ratio_schema.rb:1]: Can you add specs for this file? You can use the other schema spec files as a guide

File: spec/structuraid_core/design_codes/aci_318_19/rc/footings/bending_reinforcement_ratio_spec.rb [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        effective_height:,
+        flexural_moment:,
+        capacity_reduction_factor:
+      )
+    end
+
+    let(:design_steel_yield_strength) { 420_000_000 } # N/(m**2)
+    let(:width) { 2.5 } # m
+    let(:effective_height) { 0.45 } # m
+    let(:capacity_reduction_factor) { 0.90 }
```
Comment (mapra99) [inline spec/structuraid_core/design_codes/aci_318_19/rc/footings/bending_reinforcement_ratio_spec.rb:20]: I think our units system convention is:
```
mm for lengths
N for force
N/mm**2 (that is MPa) for stress and pressure
```

Are these specs and the formulae you coded compliant with those units?

**Review by mapra99 — COMMENTED** [review]
It looks good! I have a couple questions, and left some suggestions

File: lib/structuraid_core/design_codes/aci_318_19/rc/footings/bending_reinforcement_ratio.rb (hunk 1 above)
Comment (ImMPrada) [reply lib/structuraid_core/design_codes/aci_318_19/rc/footings/bending_reinforcement_ratio.rb:46]: Let's wrap this comment into a fix due to an error. I was thinking m, Pa, and N

File: lib/structuraid_core/design_codes/nsr_10/rc/footings/bending_reinforcement_ratio.rb (hunk 3 above)
Comment (ImMPrada) [reply lib/structuraid_core/design_codes/nsr_10/rc/footings/bending_reinforcement_ratio.rb:48]: Let's wrap this comment into a fix due to an error. I was thinking m, Pa, and N

File: spec/structuraid_core/design_codes/aci_318_19/rc/footings/bending_reinforcement_ratio_spec.rb (hunk 5 above)
Comment (ImMPrada) [reply spec/structuraid_core/design_codes/aci_318_19/rc/footings/bending_reinforcement_ratio_spec.rb:20]: fuck, you are right, so flexural moment should be N*mm

Comment (ImMPrada) [reply spec/structuraid_core/design_codes/aci_318_19/rc/footings/bending_reinforcement_ratio_spec.rb]: done!

File: lib/structuraid_core/design_codes/nsr_10/rc/footings/bending_reinforcement_ratio.rb (hunk 4 above)
Comment (ImMPrada) [reply lib/structuraid_core/design_codes/nsr_10/rc/footings/bending_reinforcement_ratio.rb:54]: done!

File: lib/structuraid_core/design_codes/aci_318_19/rc/footings/bending_reinforcement_ratio.rb (hunk 2 above)
Comment (ImMPrada) [reply lib/structuraid_core/design_codes/aci_318_19/rc/footings/bending_reinforcement_ratio.rb:53]: done!

File: lib/structuraid_core/design_codes/schemas/rc/footings/bending_reinforcement_ratio_schema.rb
Comment (ImMPrada) [reply lib/structuraid_core/design_codes/schemas/rc/footings/bending_reinforcement_ratio_schema.rb:1]: done

File: spec/structuraid_core/design_codes/schemas/rc/footings/bending_reinforcement_ratio_schema_spec.rb [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+  describe '.validate!' do
+    subject(:result) { described_class.validate!(params) }
+
+    let(:params) do
+      {
+        design_compression_strength: 28,
+        design_steel_yield_strength: 420,
+        width: 2500,
+        effective_height: 450,
+        flexural_moment: 2_348_493_750,
```
Comment (mapra99) [inline spec/structuraid_core/design_codes/schemas/rc/footings/bending_reinforcement_ratio_schema_spec.rb:13]: uugh, that's the ugly thing of using these units. Moments order of magnitude are too long

Comment (ImMPrada) [reply spec/structuraid_core/design_codes/schemas/rc/footings/bending_reinforcement_ratio_schema_spec.rb:13]: hahahha yeah!

**Pull request merged**